dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,143 @@
+package com.dotcms.rest.api.v1.template;
+
+import com.beust.jcommander.internal.Maps;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.exception.mapper.ExceptionMapperUtil;
+import com.dotcms.util.PaginationUtil;
+import com.dotcms.util.pagination.ContainerPaginator;
+import com.dotcms.util.pagination.OrderDirection;
+import com.dotcms.util.pagination.TemplatePaginator;
+import com.dotcms.uuid.shorty.ShortyIdAPI;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.VersionableAPI;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.form.business.FormAPI;
+import com.dotmarketing.portlets.templates.business.TemplateAPI;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+import org.glassfish.jersey.server.JSONP;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * CRUD of Templates
+ * @author jsanca
+ */
+@Path("/v1/templates")
+public class TemplateResource {
+
+    private final PaginationUtil paginationUtil;
+    private final WebResource    webResource;
+    private final FormAPI        formAPI;
+    private final TemplateAPI    templateAPI;
+    private final VersionableAPI versionableAPI;
```

Comment:
Codacy found an issue: [Perhaps 'versionableAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)